packages/run-protocol/test/vaultFactory/test-liquidator.js

```diff
@@ -342,19 +342,73 @@ const makeDriver = async (t, initialPrice, priceBase) => {
     E(lender).getCollateralManager(aethBrand),
   ).getNotifier();
   let managerNotification = await E(managerNotifier).getUpdateSince();
-  /** @type {UserSeat<VaultKit>} */
-  let vaultSeat;
-  /** @type {VaultKit} */
-  let vault;
+
   /** @type {UserSeat<VaultKit>} */
   let lastSeat;
-  let notifier;
   let notification = {};
   let lastOfferResult;
+  const makeVaultDriver = async (collateral, debt) => {
+    /** @type {UserSeat<VaultKit>} */
+    const vaultSeat = await E(zoe).offer(
+      await E(lender).makeVaultInvitation(),
+      harden({
+        give: { Collateral: collateral },
+        want: { RUN: debt },
+      }),
+      harden({
+        Collateral: aethMint.mintPayment(collateral),
+      }),
+    );
+    const {
+      vault,
+      publicNotifiers: { vault: notifier },
+    } = await E(vaultSeat).getOfferResult();
+    t.truthy(await E(vaultSeat).hasExited());
+    return {
+      vault: () => vault,
+      vaultSeat: () => vaultSeat,
+      notification: () => notification,
+      close: async () => {
+        lastSeat = await E(zoe).offer(E(vault).makeCloseInvitation());
+        lastOfferResult = await E(lastSeat).getOfferResult();
+        t.is(
+          lastOfferResult,
+          'your loan is closed, thank you for your business',
+        );
+        t.truthy(await E(vaultSeat).hasExited());
+      },
+      checkNotify: async (phase, expected) => {
+        notification = await E(notifier).getUpdateSince();
+        trace(t, 'notify', notification);
+        t.is(notification.value.vaultState, phase);
+        expected && t.like(notification.value, expected);
+      },
+      awaitNotify: async (phase, expected) => {
+        notification = await E(notifier).getUpdateSince(
+          notification.updateCount,
+        );
+        trace(t, 'notify', notification);
+        t.is(notification.value.vaultState, phase);
+        expected && t.like(notification.value, expected);
+      },
+      checkBorrowed: async (loanAmount, loanFee) => {
```

Comment:
the other "check" functions are void but this one gets a value. consider `getCheckedDebt` or simply `getDebt` as it's expected that the driver will be asserting all over the place.

Reply:
oh I see `checkManagerNotifier` also returns a value. same opinion on that, though that might also benefit from a signature change to remove `optUpdateSince` and have the check/await pair style you had for vault driver.

Reply:
I'm going the other way with the check/await pattern to have a single operation with all the knobs for the particular thing to check.